packages/std/src/imports.rs

```diff
@@ -148,8 +155,8 @@ impl Iterator for ExternalIterator {
     type Item = KV;
 
     fn next(&mut self) -> Option<Self::Item> {
-        let key_ptr = alloc(MAX_READ);
-        let value_ptr = alloc(MAX_READ);
+        let key_ptr = alloc(RESULT_BUFFER_LENGTH);
```

Comment:
Key_ptr can be FAR FAR less than 128KB.
I think making another constant for this and setting it to eg 2KB is more than overkill.

Reply:
Is there any specific reason to make such assumptions? I mean, an application could choose to store all its data in keys with empty values. Why not? Just because this does not happen in the average case, should we make more restrictions than necessary?

In any case, I think it is good to have separate constants. Created

```rust
/// The number of bytes of the memory region we pre-allocate for the result data in ExternalIterator.next
static DB_READ_KEY_BUFFER_LENGTH: usize = 64 * KI;
```

now to prove the point that they are independent. Happy to change the value to something small if there is a good reason to.